### PR TITLE
Refactoring ParamterizedProfileDef

### PR DESCRIPTION
**Pull request by SamuilsRulovs**
Refactored "convertIfcParameterizedProfileDef" so "convertIfcParameterizedProfileDefWithPosition" would be its sub-function

### CODE REVIEW

**Review by pjanck — APPROVED** [review]
Nicely done. This seems correct.

We need to test this with example files from the official web-documentation.

File: Core/src/IfcGeometryConverter/ProfileConverter.h [hunk 1]
```diff
@@ -1610,6 +1545,35 @@ namespace OpenInfraPlatform {
...
+				if (profileDef->Position) {
+					carve::math::Matrix transform = placementConverter->convertIfcPlacement(profileDef->Position.get());
+
+					for (std::vector<carve::geom::vector<2>> path_loop : temp_paths) {
+						for (carve::geom::vector<2> pt : path_loop) {
+							carve::geom::vector<3> pt_3d(carve::geom::VECTOR(pt.x, pt.y, 0));
+							pt_3d = transform * pt_3d;
+							pt.x = pt_3d.x;
+							pt.y = pt_3d.y;
+						}
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/ProfileConverter.h]: Are you trying to overwrite `pt` here? Was it like this also before the changes?

Becase then you would need to initialize the running variable with `&` as: `carve::geom::vector<2> &pt : path_loop`.

EDIT: Yes, it was with `&` before the change. Please be careful with such things.

**Review by pjanck — CHANGES_REQUESTED** [review]
There seem to be a mistake in applying the transformation from the position.

File: Core/src/IfcGeometryConverter/ProfileConverter.h (hunk 1 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/ProfileConverter.h]: Resolved: 5ef7ef1

**Review by pjanck — APPROVED** [review]

**Pull request merged**